File: src/test/java/br/unicamp/ic/laser/utils/PairsTest.java
package br.unicamp.ic.laser.utils;

import java.util.ArrayList;
import java.util.Arrays;

import org.junit.Assert;
import org.junit.Test;

public class PairsTest {
	@Test
    public void should_not_return_null(){
		ArrayList<ArrayList<String>> result = Utils.pairs(Arrays.asList("a", "b", "c"));

        Assert.assertNotNull(result);
    }
	
    @Test
    public void should_return_three_pairs(){
    	ArrayList<ArrayList<String>> actual = Utils.pairs(Arrays.asList("a", "b", "c"));
    	ArrayList<ArrayList<String>> expect = new ArrayList<ArrayList<String>>();

    	ArrayList<String> arrayA = new ArrayList<String>();
    	arrayA.add("a");
    	arrayA.add("b");
    	
    	ArrayList<String> arrayB = new ArrayList<String>();
    	arrayB.add("a");
    	arrayB.add("c");
    	
    	ArrayList<String> arrayC = new ArrayList<String>();
    	arrayC.add("b");
    	arrayC.add("c");

    	expect.add(arrayA);
    	expect.add(arrayB);
    	expect.add(arrayC);
    	
        Assert.assertEquals(expect, actual);
    }

    @Test
    public void should_return_empty() {
    	ArrayList<ArrayList<String>> actual = Utils.pairs(Arrays.asList());
        ArrayList<ArrayList<String>> expect = new ArrayList<ArrayList<String>>();

        Assert.assertEquals(expect, actual);
    }

    @Test
    public void should_return_two() {
    	ArrayList<ArrayList<String>> actual = Utils.pairs(Arrays.asList("0", "1"));
        ArrayList<ArrayList<String>> expect = new ArrayList<ArrayList<String>>();
        
        ArrayList<String> arrayA = new ArrayList<String>();
    	arrayA.add("0");
    	arrayA.add("1");
    	
    	expect.add(arrayA);
    	
        Assert.assertEquals(expect, actual);
    }
}

File: test/PairsTest.kt
import br.unicamp.ic.laser.main.pairs
import org.junit.Assert
import org.junit.Test

class PairsTest {
    @Test
    fun should_not_return_null(){
        val result = pairs(listOf(0, 1, 2))

        Assert.assertNotNull(result)
    }
    @Test
    fun should_return_three_pairs(){
        val result = pairs(listOf(0, 1, 2))
        val expect = listOf(listOf(0, 1), listOf(0, 2), listOf(1, 2))

        Assert.assertEquals(result, expect)
    }

    @Test
    fun should_return_empty() {
        val result = pairs(listOf<Any>())
        val expect = listOf<Any>()

        Assert.assertEquals(expect, result)
    }

    @Test
    fun should_return_two() {
        val result = pairs(listOf(0, 1))
        val expect = listOf(listOf(0 ,1))

        Assert.assertEquals(expect, result)
    }
}